cmake/modules/BuildJaeger.cmake

```diff
@@ -0,0 +1,75 @@
+###############################################################################
+#Ceph - scalable distributed file system
+#
+#Copyright (C) 2020 Red Hat Inc.
+#
+#This is free software; you can redistribute it and/or
+#modify it under the terms of the GNU Lesser General Public
+#License version 2.1, as published by the Free Software
+#Foundation.  See file COPYING.
+################################################################################
+#
+# This module builds Jaeger after it's dependencies are installed and discovered
+# opentracing: is built using cmake/modules/Buildopentracing.cmake
+# Thrift: build using cmake/modules/Buildthrift.cmake
+# yaml-cpp, nlhomann-json: are installed locally and then discovered using
+# Find<package>.cmake
+# Boost Libraries used for building thrift are build and provided by
+# cmake/modules/BuildBoost.cmake
+
+function(build_jaeger)
+  set(Jaeger_SOURCE_DIR "${CMAKE_SOURCE_DIR}/src/jaegertracing/jaeger-client-cpp")
+  set(Jaeger_INSTALL_DIR "${CMAKE_BINARY_DIR}/external")
+  set(Jaeger_BINARY_DIR "${Jaeger_INSTALL_DIR}/Jaeger")
+
+  file(MAKE_DIRECTORY "${Jaeger_INSTALL_DIR}")
+  set(Jaeger_CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+                        -DBUILD_SHARED_LIBS=ON
+                        -DHUNTER_ENABLED=OFF
+                        -DBUILD_TESTING=OFF
+                        -DJAEGERTRACING_BUILD_EXAMPLES=OFF
+                        -DCMAKE_PREFIX_PATH=${CMAKE_BINARY_DIR}/external
+                        -DCMAKE_INSTALL_RPATH=${CMAKE_BINARY_DIR}/external
+			-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
+                        -Dopentracing_DIR=${CMAKE_SOURCE_DIR}/src/jaegertracing/opentracing-cpp
+			-Dnlohmann_json_DIR=/usr/lib
+			-DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/external
+			-DCMAKE_FIND_ROOT_PATH="${CMAKE_SOURCE_DIR}/debian/tmp${CMAKE_BINARY_DIR}/external;${CMAKE_BINARY_DIR}/external"
```

Comment:
how does "debian" kick in?

Reply:
updated, should work without it, testing to verify